--- app/(dashboard)/(main)/layout.tsx
+++ app/(dashboard)/(main)/layout.tsx
@@ -21,8 +21,6 @@
 }: {
   children: React.ReactNode;
 }) {
-  throw new Error("Testing if error boundary works");
-
   const supabase = await createClient();
   const { data: { user } } = await withTimeout(
     supabase.auth.getUser(), 
@@ -28,7 +26,7 @@
     "Supabase auth.getUser() timed out!"
   );
 
-  if (!user) {
+  if (!user || !user.id) {
     redirect('/login');
   }
 
